osu.Game/Overlays/BeatmapSet/Scores/ScoresContainer.cs

```diff
@@ -59,11 +59,12 @@ protected APILegacyScores Scores
                 }
 
                 var scoreInfos = value.Scores.Select(s => s.CreateScoreInfo(rulesets)).ToList();
+                var topScore = scoreInfos.First();
 
                 scoreTable.Scores = scoreInfos;
+                scoreTable.IsBeatmapRanked = topScore.Beatmap.Status == BeatmapSetOnlineStatus.Ranked;
```

Comment:
This doesn't actually work properly for ranked maps (screenshot below from `TestSceneBeatmapSetOverlay`, `Online` step):

![image](https://user-images.githubusercontent.com/20418176/75715866-71c55e80-5cce-11ea-96a7-594e8924a339.png)

Columns are created in the setter of `Scores` one line above, so this has no effect at the point of being set to `true` here. This indicates faulty API design as now you have to call things in the right order to have them work. The solution here is to remove `Scores` and `IsBeatmapRanked` and merge them into a method:
```csharp
public void DisplayTable(IEnumerable<ScoreInfo> scores, bool showPerformanceColumn);
```
Also please add tests for that behaviour (checking for presence of the column or lack thereof) while you're at it.

Reply:
Added something similar to what you proposed - please check if I've done it correctly.

Regarding tests - those would be best suited in the `ScoreContainer` test scene which in its current state is entirely broken as noted in #7916, so it kinda hard to test anything there.